File: Java-Core/src/main/java/presto/literal/TimeLiteral.java
package presto.literal;

import org.joda.time.LocalTime;

import presto.runtime.Context;
import presto.type.IType;
import presto.type.TimeType;
import presto.value.Time;


public class TimeLiteral extends Literal<Time> {

	public TimeLiteral(String text) {
		super(text,parseTime(text.substring(1,text.length()-1)));
	}
	
	@Override
	public IType check(Context context) {
		return TimeType.instance();
	}
	
	public static Time parseTime(String text) {
		return new Time(LocalTime.parse(text));
	}
	
	
	
	
}

File: Java-CodeServer/src/main/resources/js/prompto/literal/TimeLiteral.js
var Literal = require("./Literal").Literal;
var TimeType = require("../type/TimeType").TimeType;
var Time = require("../value/Time").Time;

function TimeLiteral(text) {
	Literal.call(this, text, Time.Parse(text.substring(1,text.length-1)));
	return this;
}

TimeLiteral.prototype = Object.create(Literal.prototype);
TimeLiteral.prototype.constructor = TimeLiteral;

TimeLiteral.prototype.check = function(context) {
	return TimeType.instance;
};

exports.TimeLiteral = TimeLiteral;

